app/packs/review/_components/UnreviewedPack.tsx

```diff
@@ -8,41 +18,199 @@ import {
+				},
+				body: JSON.stringify({
+					id,
+					approved: action === 'accept' ? true : false,
+					message: action === 'reject' ? rejectionReason : undefined
+				})
+			})
+
+			if (!response.ok) throw new Error('Failed to process request')
+
+			toast({
+				title: action === 'accept' ? 'Pack Accepted' : 'Pack Rejected',
+				description:
+					action === 'accept' ?
+						'The pack has been approved and is now public'
+					:	'The pack has been rejected and the author will be notified'
+			})
+
+			setDialogState(null)
+			setRejectionReason('')
+			onStatusChange?.()
+		} catch (error) {
+			toast({
+				title: 'Error',
+				description: 'Failed to process the request',
+				variant: 'destructive'
+			})
+		} finally {
+			setIsLoading(false)
+		}
+	}
+
+	const handleDialogClose = () => {
+		// Reset rejection reason when dialog closes
+		setRejectionReason('')
+		setDialogState(null)
+	}
+
 	return (
-		<li className="p-[3px]">
-			<Card className="border-none h-full">
-				<CardHeader className="relative">
-					<CardTitle>{pack.name}</CardTitle>
-					<CardDescription>{pack.description}</CardDescription>
-				</CardHeader>
+		<Card className="relative border-none h-full flex flex-col justify-between">
+			<CardHeader>
+				<CardTitle>{name}</CardTitle>
+				<CardDescription>{description}</CardDescription>
+			</CardHeader>
+			<div>
 				<CardContent className="grid grid-cols-2">
 					<div className="flex flex-col gap-0.5">
 						<h3 className="text-sm text-muted-foreground">Questions</h3>
-						<p className="">{pack.questions.length}</p>
+						<p className="">{questions.length}</p>
 					</div>
 					<div className="flex flex-col gap-0.5">
 						<h3 className="text-sm text-muted-foreground">Type</h3>
-						<p className="">{pack.type}</p>
+						<p className="">{packMap[type as PackType]}</p>
 					</div>
 				</CardContent>
-				<CardFooter>
-					<UnreviewedPackDetails {...pack} />
+				<CardFooter className="grid grid-cols-2 gap-2 mt-auto h-fit">
+					<AlertDialog
+						open={dialogState === 'accept'}
+						onOpenChange={(open) => !open && handleDialogClose()}
+					>
+						<AlertDialogTrigger asChild>
+							<Button
+								variant="default"
+								className="bg-emerald-500 hover:bg-emerald-600 text-white"
+								disabled={isLoading}
+								onClick={() => setDialogState('accept')}
+							>
+								<Check className="mr-2 h-4 w-4" />
+								Accept
+							</Button>
+						</AlertDialogTrigger>
+						<AlertDialogContent>
+							<AlertDialogHeader>
+								<AlertDialogTitle>Accept pack?</AlertDialogTitle>
+								<AlertDialogDescription>
+									Accepting this pack will make it public and available to all
+									users. Are you sure you want to proceed?
+								</AlertDialogDescription>
+							</AlertDialogHeader>
+							<AlertDialogFooter>
+								<AlertDialogCancel>Cancel</AlertDialogCancel>
+								<Button
+									onClick={() => handleAction('accept')}
+									className="text-white gap-3"
```

Comment:
nit: text-white is not needed, its set by the default variant
```suggestion
									className="gap-3"
```